staging/src/k8s.io/endpointslice/reconciler.go

```diff
@@ -275,8 +281,15 @@ func (r *Reconciler) reconcileByAddressType(logger klog.Logger, service *corev1.
 			}
 			r.topologyCache.RemoveHints(si.ServiceKey, addressType)
 		}
-		slicesToCreate, slicesToUpdate = topologycache.RemoveHintsFromSlices(si)
+		if !canUseTrafficDistribution {
+			slicesToCreate, slicesToUpdate = topologycache.RemoveHintsFromSlices(si)
+		}
+	}
+
+	if canUseTrafficDistribution {
+		slicesToCreate, slicesToUpdate, _ = trafficdist.ReconcileHints(service.Spec.TrafficDistribution, slicesToCreate, slicesToUpdate, unchangedSlices(existingSlices, slicesToUpdate, slicesToDelete))
```

Comment:
The if block from line 270 to 290 is super hard to read, 

so `if utilfeature.DefaultFeatureGate.Enabled(features.TopologyAwareHints)`then `topologyCache`will be not nil.

If TopologyAwareHints then we have to check
- if hints is enabled, if not
- if traffic distribution is enabled, if not
- removehints

it seems we can move this to a swtich statement or to organize this in another way ( and with comments) so we can follow better what is happening, WDYT?

Reply:
oh, so the whole block with topologyCache goes away when we deprecate the annotation, right?

then is ok, scratch this comment